utils/PathSanitizingFileCheck

```diff
@@ -67,6 +67,28 @@ constants.""")
         slashes_re = r'/'
 
     stdin = sys.stdin.read()
+
+    if args.enable_windows_compatibility:
+        # Let's look for paths in the output, and try to transform them to use
+        # Unix directory separators to be automatically picked up by the tests.
+        # This only picks up absolute paths, but those should be the more
+        # common in the tools output and the easier to detect without false
+        # positives.
+
+        def replace_slashes(matchobj):
+            return re.sub(r'\\\\|\\', r'/', matchobj.group(0))
+
+        # The regex is composed of three parts:
+        # \b[a-zA-Z]:(?:\\\\|\\|\/) -> Matches a drive letter followed by a
+        #                              slash (backward escaped, backward or
+        #                              forward)
+        # (?:[-a-zA-Z0-9_.]+(?:\\\\|\\|\/))* -> Matches the path part, it always
+        #                                       ends up in a slash.
+        # (?:[-a-zA-Z0-9_.]+)?\b -> Matches the last path component, which do
+        #                           not has a trailing slash, but it is
+        #                           optional.
+        stdin = re.sub(r'\b[a-zA-Z]:(?:\\\\|\\|\/)(?:[-a-zA-Z0-9_.]+(?:\\\\|\\|\/))*(?:[-a-zA-Z0-9_.]+)?\b', replace_slashes, stdin)
```

Comment:
Yikes. @compnerd?

Reply:
Yikes indeed.

Saleem has already changed a lot of tests with those `{{\\\\|/}}`, but it makes reading the test files very complicated (IMO). This hopefully will avoid having to add any more of those, and people that work on Unixes will not have to care about Windows. I hope Saleem sees the benefits, but rest assure that I flagged myself after writing this.

Reply:
I, I .... words... I.  If it works? 🤷‍♂